ui/pages/notifications/snap/types/types.ts

```diff
@@ -0,0 +1,17 @@
+export const SNAP = 'SNAP' as const;
+
+export type RawSnapNotification = {
+  id: string;
+  message: string;
+  origin: string;
+  createdDate: number;
+  readDate?: number;
+};
+
+export type SnapNotification = {
+  id: string;
+  createdAt: string;
+  isRead: boolean;
+  type: typeof SNAP;
+  data: RawSnapNotification;
+};
```

Comment:
This is the new snap notification shape.

The `data` field retains the original snap shape.

The reason for these additional fields are for Notification interface consistency & so we can figure out which notification is which (via the `type` field acting as a discriminated union.)